java-checks/src/main/resources/org/sonar/l10n/java/rules/java/S5852_java.html

```diff
@@ -14,21 +14,55 @@ <h2>Ask Yourself Whether</h2>
-  nested quantifier situation <code>(ba+)+</code> doesn't cause performance issues, indeed, the inner group can be matched only if it exists exactly
-  one <code>b</code> char per repetition of the group. </li>
-  <li> Make sure that consecutive repetitions can't match the same contents or are separated by a non-optional separator that can't be matched by both
-  of the repetitions. So for example <code>a*b*</code> is not a problem because <code>a*</code> and <code>b*</code> match different things and
-  <code>a*_a*</code> is not a problem because the repetitions are separated by a '_' and can't match that '_'. However, <code>a*a*</code> and
-  <code>.*_.*</code> have quadratic runtime. </li>
+  nested quantifier situation <code>(ba+)+</code> doesn't cause performance issues, indeed, the inner group can be matched only if there exists
+  exactly one <code>b</code> char per repetition of the group. </li>
   <li> Optimize regular expressions with <code>possessive quantifiers</code> and <code>atomic grouping</code>. </li>
   <li> Use negated character classes instead of <code>.</code> to exclude separators where applicable. For example the quadratic regex
   <code>.*_.*</code> can be made linear by changing it to <code>[^_]*_.*</code> </li>
 </ul>
+<p>Sometimes it's not possible to rewrite the regex to be linear while still matching what you want it to match. Especially when using partial
+matches, for which it is quite hard to avoid quadratic runtimes. In those cases consider the following approaches:</p>
+<ul>
+  <li> Solve the problem without regular expressions </li>
+  <li> Use an alternative non-backtracking regex implementations such as Google's <a href="https://github.com/google/re2">RE2</a> / <a
+  href="https://github.com/google/re2j">RE2/J</a>. </li>
+  <li> Use multiple passes. This could mean pre- and/or post-processing the string manually before/after applying the regular expression to it or
+  using multiple regular expressions. One example of this would be to replace <code>str.split("\\s*,\\s*")</code> with <code>str.split(",")</code> and
+  then trimming the spaces from the strings as a second step. </li>
+  <li> When using `Matcher.find()` it is often possible to make the regex infallible by making all the parts that could fail optional, which will
+  prevent backtracking. Of course this means that you'll accept more strings than intended, but this can be handled by using capturing groups to check
+  whether the optional parts were matched or not and then ignoring the match if they weren't. For example the regex <code>x*y</code> could be replaced
+  with <code>x*\(y)?</code> and then the call to <code>matcher.find()</code> could be replaced with <code>matcher.find() &amp;&amp; matcher.group(1)
+  != null</code>. </li>
+</ul>
 <h2>Sensitive Code Example</h2>
 <p>The first regex evaluation will never end in <code>OpenJDK</code> &lt;= 9 and the second regex evaluation will never end in any versions of
 <code>OpenJDK</code>:</p>
```

Comment:
We should update the Java specific part to mention `JDK` over `OpenJDK` for consistency